tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -254,14 +329,47 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
                            m::Convert(m::CustomCall(&existing_gemm,
                                                     {kGemmCallTarget,
                                                      kCublasLtMatmulCallTarget})
-                                          .WithElementType(BF16)),
-                           m::Convert(m::Op(&bias).WithElementType(BF16))))
+                                          .WithOneUser()
+                                          .WithElementType(BF16))
+                               .WithOneUser(),
+                           m::Convert(m::Op(&bias).WithElementType(BF16))
+                               .WithOneUser())
+                           .WithOneUser())
                 .WithElementType(BF16))) {
       return FuseMatrixBiasAdd(instr, bias, existing_gemm);
     }
     return OkStatus();
   }
 
+  // Replaces binary(slice/bitcast(gemm), broadcast) with
+  // slice/bitcast(binary(gemm, broadcast)) and changes the shape of broadcast
+  // from that of slice/bitcast to that of the GEMM, i.e. the operand of
+  // slice/bitcast.
+  Status SinkSliceOrBitcastBelowBinaryOp(HloInstruction *slice_or_bitcast,
+                                         HloInstruction **binary,
+                                         HloInstruction **broadcast) {
+    TF_RET_CHECK(slice_or_bitcast->user_count() == 1);
+    TF_RET_CHECK((*broadcast)->user_count() == 1);
+    TF_RET_CHECK((*binary)->IsRoot() || (*binary)->user_count() == 1);
```

Comment:
What check beforehand guarantees that this is true? I am seeing a test fail due to user_count being 2.

Should `m::MaximumAnyOrder` in `HandleMaximum` add an additional `WithOneUser()`?

Reply:
In fact, is there an issue if `binary` has multiple users? Wont `binary` be replaced with the new fused instruction? Then all the users will have the new fused instructions as input.

Reply:
It seems that the `TF_RET_CHECK` in line 353 is too strict and can be removed. The maximum instruction, i.e. `*binary`, can be allowed to have multiple users since the fused GEMM produces an equivalent result.

Reply:
Thanks for confirming, @philipphack . I will make this change.